### ISSUE TITLE
`$PSStyle.Formatting.TableHeader` writes out the ANSI codes in a legacy console that doesn't support virtual terminal

### ISSUE THREAD

**Issue by daxian-dbw** [PowerShell/PowerShell#15978]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Start PowerShell 7.2.0-preview.9 with a legacy console host (e.g. use 2012R2), and then run `Get-Module`.

Expected: The table view should not have any ANSI code written out.

Actual: ANSI codes are written out in the table view

![issue-preview 9](https://user-images.githubusercontent.com/127450/130497524-a53fe0c0-cb74-4860-aba0-9df2079f8d26.JPG)

### Expected behavior

```console
The table view should not have any ANSI code written out.
ANSI code should be ignored when the host doesn't support virtual terminal.
```

### Actual behavior

```console
ANSI codes are written out in the table view
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.0-preview.9
PSEdition                      Core
GitCommitId                    7.2.0-preview.9
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by ertyz** [comment]
Is it ok on Windows 2016 LTS or not?

**Comment by Pekka-Kalevi** [comment]
Should ANS.1 characters be filtered by Out-Default depending current host?

**Comment by daxian-dbw** [comment]
@ertyz I didn't try but I guess it will have the same issue on Windows Server 2016. But we will get it fixed before 7.2 GA

@Pekka-Kalevi It should be filtered out depending on current host.
